### PR TITLE
Add module and kernel for Domain Cached Credentials (DCC), MS Cache (NT), -m 31500

### PR DESCRIPTION
**Pull request by Chick3nman**
Adds module and kernel for Domain Cached Credentials (DCC), MS Cache (NT), -m 31500. This is a specialist kernel/module for hash shucking using NT hashes.

Draft PR due to missing test .pm for the module.

### CODE REVIEW

**Review by Michael-soft-216 — APPROVED** [review]
ok

**Comment by Chick3nman** [comment]
Unit test added in PR #3725

File: src/modules/module_31500.c [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+  return tmp_size;
+}
+
+
+int module_hash_decode (MAYBE_UNUSED const hashconfig_t *hashconfig, MAYBE_UNUSED void *digest_buf, MAYBE_UNUSED salt_t *salt, MAYBE_UNUSED void *esalt_buf, MAYBE_UNUSED void *hook_salt_buf, MAYBE_UNUSED hashinfo_t *hash_info, const char *line_buf, MAYBE_UNUSED const int line_len)
+{
+  u32 *digest = (u32 *) digest_buf;
+
+  hc_token_t token;
+
```
Comment (matrix) [inline src/modules/module_31500.c:70]: @Chick3nman please add here this:

memset (&token, 0, sizeof (hc_token_t));

**Comment by matrix** [comment]
Hi :) if you put the missing memset() it's ready to merge. The same also for #3657

**Comment by Chick3nman** [comment]
I've added the memset for both this one and the other one. I had missed the original comments, apologies for the delay.

File: src/modules/module_31500.c (hunk 1 above)
Comment (Chick3nman) [reply src/modules/module_31500.c:70]: Added

**Pull request merged**